### PR TITLE
EOS-18924 Code changes and code review comment fixes for handling persistent messages in dtm0

### PR DESCRIPTION
**Pull request by mehjoshi**
There is a change in the checkin branch for dtm0 post merge. This PR is opened in continuation and to apply the changes and review comments from https://github.com/Seagate/cortx-motr/pull/632

### CODE REVIEW

**Review by ivan-alekhin — APPROVED** [review]
Overall the patch looks good. I added a minor comment (we can address that later).

File: dtm0/service.c [hunk 1]
```diff
@@ -19,6 +19,7 @@
  *
  */
 
+#include <asm-generic/errno.h>
```
Comment (just-now) [inline dtm0/service.c:22]: This can be safely removed.

File: dtm0/fop.c [hunk 2]
```diff
@@ -307,55 +309,97 @@ M0_INTERNAL int m0_dtm0_logrec_update(struct m0_be_dtm0_log  *log,
...
-	rc = m0_dtm0_tx_desc_copy(txd, &msg);
-	if (rc != 0)
-		return M0_ERR(rc);
 
+	M0_PRE(log != NULL);
 	/*
-	 * TODO: This change will be done by DTM0 log because
-	 * it should log the entry "pa == self" with PERSISTENT state
-	 * set.
+         * TODO: uncomment the following line after the connections code
```
Comment (sergey-shilov) [inline dtm0/fop.c:326]: Indentation.

File: dtm0/fop.c [hunk 3]
```diff
@@ -307,55 +309,97 @@ M0_INTERNAL int m0_dtm0_logrec_update(struct m0_be_dtm0_log  *log,
...
+			FID_P(&txd.dtd_ps.dtp_pa->p_fid));
+	/* Send P msgs to the rest of the participants. */
+	for (i = 0; i < txd.dtd_ps.dtp_nr; ++i) {
+		if (m0_fid_eq(&txd.dtd_ps.dtp_pa[i].p_fid,
+			      &dtms->dos_generic.rs_service_fid))
+			continue;
+		rc = m0_dtm0_send_msg(fom, DTM_PERSISTENT,
+				       &txd.dtd_ps.dtp_pa[i].p_fid, &txd);
+		if (rc != 0)
+			rc = M0_ERR_INFO(rc, "failed to send PERSISTENT msg "
```
Comment (sergey-shilov) [inline dtm0/fop.c:358]: We don't need a reassignment here, it is needed in situation like that

`		rc = M0_ERR_INFO(-ENOENT, "Failed to get pool version "FID_F,
				 FID_P(&cr->cr_pver));`

**Review by sergey-shilov — COMMENTED** [review]
Just a couple of minor comments.

File: dtm0/service.c [hunk 4]
```diff
@@ -250,9 +251,10 @@ static int dtm0_process_disconnect(struct dtm0_process *process)
 	rc = m0_rpc_link_is_connected(&process->dop_rlink) ?
 		m0_rpc_link_disconnect_sync(&process->dop_rlink, timeout) : 0;
 
-	if (M0_IN(rc, (0, -ETIMEDOUT))) {
-		if (rc == -ETIMEDOUT) {
-			M0_LOG(M0_WARN, "Disconnect timeout (suppressed)");
+	if (M0_IN(rc, (0, -ETIMEDOUT, -ECANCELED))) {
+		/* TODO: Fix this. We are ignoring -ECANCELED for now.*/
+		if (rc == -ETIMEDOUT || rc == -ECANCELED) {
+			M0_LOG(M0_WARN, "Disconnect timeouted or cancelled (suppressed)");
```
[the remark this answers is not in the document]
Comment (mehjoshi) [reply dtm0/service.c:257]: OK. Added this change.

File: dtm0/service.c (hunk 1 above)
Comment (mehjoshi) [reply dtm0/service.c:22]: Yes, removed it. Thanks.

File: dtm0/fop.c (hunk 2 above)
Comment (mehjoshi) [reply dtm0/fop.c:326]: Thanks, done.

File: dtm0/fop.c (hunk 3 above)
Comment (mehjoshi) [reply dtm0/fop.c:358]: Done. Fixed it.

**Review by just-now — APPROVED** [review]
I looked through the code. It looks fine to land to dtm0-main

**Pull request merged**